src/components/AnonymousChat/child/ChatItemAttachment.tsx

```diff
@@ -0,0 +1,109 @@
+          .map((media) => ({url: media.asset_url}))
+      });
+    }
+  };
+
+  return (
+    <View
+      style={[
+        ChatItemAttachmentStyles.attachmentContainer,
+        attachments?.find((item) => item.type === 'file') ? {height: 'auto'} : {}
+      ]}>
+      {attachments
+        .filter((item, index) => index <= 3)
+        .map((item, index) =>
+          item.type === 'file' ? (
+            <TouchableOpacity
+              key={item.file_path}
+              style={{flex: 1}}
+              activeOpacity={1}
+              onPress={() => Linking.openURL(item.file_path)}>
+              <View style={ChatItemAttachmentStyles.attachmentFileContainer}>
+                <View style={ChatItemAttachmentStyles.attachmentFileContent}>
+                  <Text style={ChatItemAttachmentStyles.attachmentFileName} numberOfLines={1}>
+                    {item.file_name}
+                  </Text>
+                  <View>
+                    <Text style={ChatItemAttachmentStyles.attachmentFileInfo}>
+                      {formatBytes(item.file_size)} •{' '}
+                      {item.file_name
+                        ?.split('.')
+                        [item.file_name?.split('.')?.length - 1]?.toUpperCase()}
+                    </Text>
+                  </View>
+                </View>
+                <View style={ChatItemAttachmentStyles.attachmentFileIcon}>
+                  <IconFile />
+                </View>
+              </View>
+            </TouchableOpacity>
+          ) : (
+            <TouchableOpacity
+              key={item.thumb_url}
+              style={{
+                width: `${
+                  (attachments.length >= 3 && index > 0) || attachments.length >= 4 ? 50 : 100
+                }%`,
+                height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
+                position: 'relative',
+                overflow: 'hidden'
+              }}
+              activeOpacity={1}
+              onPress={
+                attachments.length > 0 && item.type !== 'gif'
+                  ? () => onOpenMediaPreview(attachments, index)
+                  : null
+              }>
+              {item.type !== 'file' && (
+                <FastImage
+                  style={ChatItemAttachmentStyles.image}
+                  source={{
+                    uri: item.thumb_url
+                  }}
+                />
+              )}
+              {attachments.length > 4 && index === 3 && (
+                <View style={ChatItemAttachmentStyles.moreOverlay}>
+                  <Text style={ChatItemAttachmentStyles.moreText}>+{attachments.length - 4}</Text>
+                </View>
+              )}
+              {/* Video Play Icon */}
+              {item.video_path && (
+                <View style={ChatItemAttachmentStyles.moreOverlay}>
+                  <IconVideoPlay />
+                </View>
+              )}
+            </TouchableOpacity>
+          )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the file attachments, the code for rendering image and video attachments is also complex and could be extracted into a separate component.



<!-- This is an auto-generated comment by OSS CodeRabbit -->